server/src/test/java/org/opensearch/cluster/metadata/MetadataCreateIndexServiceTests.java

```diff
@@ -1217,6 +1218,26 @@ public void testvalidateIndexSettings() {
         threadPool.shutdown();
     }
 
+    public void testIndexTemplateReplicationType() {
+        Settings templateSettings = Settings.builder().put(INDEX_REPLICATION_TYPE_SETTING.getKey(), ReplicationType.SEGMENT).build();
```

Comment:
If you want to be extra thorough here you could assert that the default value of the `CLUSTER_REPLICATION_TYPE_SETTING` constant is not `SEGMENT` so you guarantee this test is not relying on the default value (if that were ever to change in the future)